apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -3443,7 +3443,11 @@ pub(crate) fn compute_nodes_for_tree(
             }
         }
     }
-    snapshot!(dependency_graph, "updated_dependency_graph");
+    snapshot!(
+        "FetchDependencyGraph",
+        dependency_graph.to_dot(),
+        "Fetch dependency graph updated by compute_nodes_for_tree"
+    );
```

Comment:
I'm mostly leaving this as a note about what I've described before. In the viz tool, we pair the `FetchDependencyGraph` variant of the snapshot kind with visualizing the data with graph viz. I don't love the leaning into the already string-ness of this API. Perhaps this requires rethinking how the snapshot macro works.

Perhaps, we should have a series of macros that will emit logs with the correct data? For example, this could be replaced with a `graph_snapshot!`, which would add the correct `data` field to the log (which would note how to visualize it rather than using the type) and then call `.to_dot()` on the `dependency_graph`.

I'm not asking for this to change in the PR. More, I'd like to know your thoughts.

Reply:
Good point. String literals like `"FetchDependencyGraph "` are brittle and we need a better system. Maybe some structs (and a trait) to drive this snapshot generation and type tagging.

I've added a future direction section in our hackathon doc.